scripts/bisect-test.ts

```diff
@@ -0,0 +1,48 @@
+/// <reference path="..\src\harness\external\node.d.ts" />
+
+import cp = require('child_process');
+import fs = require('fs');
+
+var args = process.argv.slice(2);
+
+var jake = cp.exec('jake clean local', () => void 0);
+jake.on('close', code => {
+    if (code === 0) {
+        // See what we're being asked to do
+        if (args[1] === 'compiles' || args[1] === '!compiles') {
+            var tsc = cp.exec('node built/local/tsc.js ' + args[0], () => void 0);
+            tsc.on('close', tscCode => {
+                if ((tscCode === 0) === (args[1] === 'compiles')) {

```

Comment:
How unexpressive - this is why we need to support the boolean NXOR operator `!^^`.